rero_mef/authorities/mef/models.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of RERO MEF.
+# Copyright (C) 2018 RERO.
+#
+# RERO MEF is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# RERO MEF is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with RERO MEF; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, RERO does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Define relation between records and buckets."""
+
+from __future__ import absolute_import
+
+from invenio_db import db
+from invenio_pidstore.models import RecordIdentifier
+from invenio_records.models import RecordMetadataBase
+
+
+class MefIdentifier(RecordIdentifier):
+    """Sequence generator for mef Authority identifiers."""
+
+    __tablename__ = 'mef_id'
+    __mapper_args__ = {'concrete': True}
+
+    recid = db.Column(
+        db.BigInteger().with_variant(db.Integer, 'sqlite'),
+        primary_key=True,
+        autoincrement=True,
+    )
+
+
+class MefMetadata(db.Model, RecordMetadataBase):
+    """Represent a record metadata."""
+
+    __tablename__ = 'mef_metadata'
+
+
+class MefAction(object):
+    """Class holding all availabe Mef record creation actions."""
+
+    CREATE = 'create'
+    UPDATE = 'update'
+    UPTODATE = 'uptodate'
+    DISCARD = 'discard'
+    DELETE = 'delete'
+    DELETEMEF = 'deletemef'
```

Comment:
Maybe better to use as an `Enum` class ? https://docs.python.org/3/library/enum.html

Reply:
This is old code, maybe I can change.